Examples/ImageRegistrationMethodBSpline2/ImageRegistrationMethodBSpline2.cxx

```diff
@@ -0,0 +1,139 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+// A SimpleITK example demonstrating image registration using the
+// BSplineTransform and the MattesMutualInformation metric.
+
+#include <SimpleITK.h>
+#include <iostream>
+#include <stdlib.h>
+#include <iomanip>
+
+namespace sitk = itk::simple;
+
+class IterationUpdate
+{
+public:
+  IterationUpdate(const sitk::ImageRegistrationMethod & m)
+    : m_Method(m)
+  {}
+
+  void
+  Execute()
+  {
+    std::cout << std::setw(3) << m_Method.GetOptimizerIteration();
+    std::cout << " = " << std::setw(10) << std::setprecision(5) << m_Method.GetMetricValue() << std::endl;
+    std::cout << "\t#: " << m_Method.GetOptimizerPosition().size() << std::endl;
+  }
+
+private:
+  const sitk::ImageRegistrationMethod & m_Method;
+};
+
+class MultiResolutionIterationUpdate
+{
+public:
+  MultiResolutionIterationUpdate(const sitk::ImageRegistrationMethod & m)
+    : m_Method(m)
+  {}
+
+  void
+  Execute()
+  {
+    std::cout << "--------- Resolution Changing ---------" << std::endl;
+  }
+
+private:
+  const sitk::ImageRegistrationMethod & m_Method;
+};
```

Comment:
It likely would be more convenient to implement this a direct lambda call.

Reply:
Done and force pushed.